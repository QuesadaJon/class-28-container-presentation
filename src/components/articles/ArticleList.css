.ArticleList {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  align-items: center;
  list-style: none;
}