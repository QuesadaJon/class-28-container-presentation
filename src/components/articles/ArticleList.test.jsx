import React from 'react';
import { render } from '@testing-library/react';
import ArticleList from './ArticleList';

const articles = [
  {
    author: 'https://image.com',
    image: 'image',
    title: 'title1',
    description: 'description',
    url: 'url'
  },
  {
    author: 'https://image.com',
    image: 'image',
    title: 'title2',
    description: 'description',
    url: 'url'
  },
  {
    author: 'https://image.com',
    image: 'image',
    title: 'title3',
    description: 'description',
    url: 'url'
  },
  {
    author: 'https://image.com',
    image: 'image',
    title: 'title4',
    description: 'description',
    url: 'url'
  }
];

describe('ArticleList component', () => {
  it('matches a snapshot', () => {
    const { asFragment } = render(
      <ArticleList articles={articles} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
})
;
