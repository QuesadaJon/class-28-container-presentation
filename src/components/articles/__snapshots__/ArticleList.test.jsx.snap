// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component matches a snapshot 1`] = `
<DocumentFragment>
  <ul
    class="ArticleList"
    data-testid="articles"
  >
    <li>
      <a
        class="Article"
        href="url"
      >
        <figure>
          <img
            alt="title1"
            src="image"
          />
          <figcaption>
            title1
          </figcaption>
          <p>
            https://image.com
          </p>
          <p>
            description
          </p>
        </figure>
      </a>
    </li>
    <li>
      <a
        class="Article"
        href="url"
      >
        <figure>
          <img
            alt="title2"
            src="image"
          />
          <figcaption>
            title2
          </figcaption>
          <p>
            https://image.com
          </p>
          <p>
            description
          </p>
        </figure>
      </a>
    </li>
    <li>
      <a
        class="Article"
        href="url"
      >
        <figure>
          <img
            alt="title3"
            src="image"
          />
          <figcaption>
            title3
          </figcaption>
          <p>
            https://image.com
          </p>
          <p>
            description
          </p>
        </figure>
      </a>
    </li>
    <li>
      <a
        class="Article"
        href="url"
      >
        <figure>
          <img
            alt="title4"
            src="image"
          />
          <figcaption>
            title4
          </figcaption>
          <p>
            https://image.com
          </p>
          <p>
            description
          </p>
        </figure>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
